### ISSUE TITLE
Add access method for formatted body to MatrixJsonRoomMessageEvent

### ISSUE THREAD

**Issue by MrCustomizer** [kamax-io/matrix-java-sdk#33]
Room message events can contain formatted text. Several clients also seem to more or less use this feature. Especially Riot makes excessive use of it. I stumbled upon it when learning about the mechanisms of how Riot handles mentions. [Mention pills are html link tags, where href follows a certain formatting standard.](https://docs.google.com/document/d/1oRhw3DJhbVAKkHAEgyt6ccV82wtXR_11qY7JqMcesUU/edit)

[Several other html tags for formatting text are also supported.](https://docs.google.com/document/d/1QPncBmMkKOo6_B2jyBuy5FFSZJrRsq7WU5wgRSzOMho/edit)

Long story short: `MatrixJsonRoomMessageEvent` should be extended, so that clients are able to access the formatted text body sent by other clients instead of only the raw text body. I propose adding the methods `getFormattedBody` and `getFormat` to access the JSON fields `formatted_body` and `format` in the `content` object of the event.

**Comment by MrCustomizer** [comment]
Some kind of check-method to determine, if a formatted body is present is probably also needed as clients are always allowed to only send raw text.
